service/geopmdpy/system_files.py

```diff
@@ -49,6 +52,32 @@
 """
 
 
+@lru_cache(None)
```

Comment:
Looks like this decorator is what enables the single write of the warning message.

Reply:
That's right. Since `get_config_path` takes no args, the decorator ensures this function's body is only executed once.